### PR TITLE
Replace SCC strategic merge patch with JSON patch.

### PR DESCRIPTION
**Pull request by davidvossel**
**What this PR does / why we need it**:

Some versions of OpenShift don't support StrategicMerge patch type for SCC objects. This prevents virt-operator from installing kubevirt on those versions of OpenShift. 

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1709677

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/install-strategy/create.go [hunk 1]
```diff
@@ -1392,13 +1394,19 @@ func addOrRemoveSSC(targetStrategy *InstallStrategy,
...
+			}
 			userBytes, err := json.Marshal(users)
 			if err != nil {
 				return err
 			}
 
-			data := []byte(fmt.Sprintf(`{"users": %s}`, userBytes))
-			_, err = scc.SecurityContextConstraints().Patch(sccPriv.TargetSCC, types.StrategicMergePatchType, data)
+			test := fmt.Sprintf(`{ "op": "test", "path": "/users", "value": %s }`, string(oldUserBytes))
+			patch := fmt.Sprintf(`{ "op": "replace", "path": "/users", "value": %s }`, string(userBytes))
```
Comment (stu-gott) [inline pkg/virt-operator/install-strategy/create.go]: Is it possible to ever have an empty or omitted `/users` struct? I've found that `replace` can error unexpectedly if e.g. `/users` was missing entirely. in which case `add` needs to be used instead (and no test is needed as there's an implicit "and /users must not already exist")

File: pkg/virt-operator/install-strategy/delete.go [hunk 2]
```diff
@@ -278,24 +278,33 @@ func DeleteAll(kv *v1.KubeVirt,
...
+				return err
+			}
 			userBytes, err := json.Marshal(users)
 			if err != nil {
 				return err
 			}
 
-			data := []byte(fmt.Sprintf(`{"users": %s}`, userBytes))
-			_, err = scc.SecurityContextConstraints().Patch(sccPriv.TargetSCC, types.StrategicMergePatchType, data)
+			test := fmt.Sprintf(`{ "op": "test", "path": "/users", "value": %s }`, string(oldUserBytes))
```
Comment (cynepco3hahue) [inline pkg/virt-operator/install-strategy/delete.go]: Just curious what exactly `test` operator is doing?

Comment (stu-gott) [reply pkg/virt-operator/install-strategy/delete.go]: @cynepco3hahue JSON patches have the ability to issue a `test` command to ensure the contents of the resource match an expected state before continuing. this allows the abilty to do something like "test the users = [bob]" and the "replace the contents of users with [bob, alice]" ... and since the json patch is applied all at once this all happens in one step. 

if you just did the replace command without checking the contents you could accidentally clobber a conflicting change.

Comment (davidvossel) [reply pkg/virt-operator/install-strategy/delete.go]: We don't own the SCC object, but our PATCH is attempting to replace the entire "users" array.

The "test" ensures that we aren't colliding with another process that's also attempting to modify "users".  Test means that if "users" isn't exactly equal to what we think it shoudl be equal to before the "replace" that the Patch operation should fail.

so, if another process sneaks in and adds a user to the users array right before our Patch operation, the "test" op would fail causing the Patch to fail. 

Basically "test" is a safety mechanism.

File: pkg/virt-operator/install-strategy/create.go (hunk 1 above)
Comment (stu-gott) [reply pkg/virt-operator/install-strategy/create.go]: Confirmed to be a non-issue

Comment (davidvossel) [reply pkg/virt-operator/install-strategy/create.go]: Good point. It looks like we're okay in this instance though.

I tested this by creating an SCC rule that omitted the 'users' array in the yaml file. Then I patched that SCC object with the "replace" op. It worked.

File: pkg/virt-operator/install-strategy/delete.go (hunk 2 above)
Comment (cynepco3hahue) [reply pkg/virt-operator/install-strategy/delete.go]: Thanks for detailed answer
